### PR TITLE
BosTau edits

### PR DESCRIPTION
**Pull request by gregorgorjanc**
This PR addresses #601 and #606

### CODE REVIEW

**Comment by grahamgower** [comment]
Looks good @gregorgorjanc! CircleCI is complaining about some formatting issues (caught by the `flake8` command). If you can fix those, then do a `git rebase` to squash the commits, this should be good to merge. (There's information about this process in the [dev docs](https://stdpopsim.readthedocs.io/en/latest/development.html) should you need it)

**Comment by gregorgorjanc** [comment]
@grahamgower I tried to address the flake8 suggestions but then got bitten by git rebase on squashing because I tried to rebase to a commit that was already pushed. Hence so many commits. Not fun. I still haven't sorted all flake8 suggestions and now linting is failing me. Can you guide me, please?

**Comment by ndukler** [comment]
> @grahamgower I tried to address the flake8 suggestions but then got bitten by git rebase on squashing because I tried to rebase to a commit that was already pushed. Hence so many commits. Not fun. I still haven't sorted all flake8 suggestions and now linting is failing me. Can you guide me, please?

After you do the interactive rebase you need to force push. If you look at the dev docs and search for the term force it will give you an example.

File: stdpopsim/catalog/BosTau/__init__.py [hunk 1]
```diff
@@ -127,64 +126,51 @@ def _HolsteinFriesan_1M13():
...
-            ),  # Ne 17000 to 62000
-            msprime.PopulationParametersChange(
-                time=33155, growth_rate=-0.0000, population_id=0
-            ),  # Ne 62000 (model has "coalesced")
-            msprime.PopulationParametersChange(
-                time=933155, growth_rate=-0.0, population_id=0
-            ),
+            #      3 generations at   120,     4-     6
+            msprime.PopulationParametersChange(time=4, initial_size=120,
+                population_id=0),
```
Comment (grahamgower) [inline stdpopsim/catalog/BosTau/__init__.py:138]: flake8 is complaining here: `E128 continuation line under-indented for visual indent`. What this means, is that the `time=...` bit and the `population_id=...` bits are not indented to the same column. I agree this is quite frustrating.

**Comment by grahamgower** [comment]
Regarding the code formatting, one thing you can try is to reformat the file using the [black](https://github.com/psf/black) code formatting tool. Just `pip install black`, or whatever you usually do to install python packages, and then run `black stdpopsim/catalog/BosTau/__init__.py`, and add/commit the changes. We intend to make reformatting with black a mandatory part of the dev process in the future (#433), but I haven't spent the time to understand how we make that easy for developers (tskit and msprime repositories use yet *another* tool to manage this, called [pre-commit](https://pre-commit.com/)).

Regarding rebasing and squashing, @ndukler rightly points out that there's a section on this in the dev docs https://stdpopsim.readthedocs.io/en/latest/development.html#github-workflow
I'm not sure exactly what you meant by trying to "rebase to a commit that was already pushed", but to avoid issues you should definitely do a `git fetch upstream` (it's possible this is called something different to 'upstream' - you can check what remote repositories are configured in your local repository with `git remote -v`) and then explicitly specify the remote branch that you're rebasing against (e.g. `git rebase -i upstream/master`).

**Comment by grahamgower** [comment]
Hey @gregorgorjanc, this looks like it's basically ready to go. Just some code format issues causing the CI failure. Our code-formatting process and CI tests have changed in the past couple of weeks. There are [updated instructions](https://stdpopsim.readthedocs.io/en/latest/development.html#github-workflow) about using `pre-commit`, and this should automatically do the formatting for you. Are you comfortable with rebasing/squashing and trying out the new pre-commit procedure? There might be some merge issues though (a bunch of stuff has changed, all over the code base), so let me know if you'd prefer me to deal with it instead.

**Comment by gregorgorjanc** [comment]
See #809 - not sure how I managed to duplicate PR. Sorry.

**Pull request closed without merging**